### ISSUE TITLE
Unstable formatter adds extra indentation on every execution

### ISSUE THREAD

**Issue by logicalChimp** [Riverside-Healthcare/djLint#367]
The following sample was tested with v1.9.4 and 1.12.1 - both exhibit the same behaviour (project is running 1.9.4, and 1.12.1 is latest version at the time this issue is being raised)

Issue #166 appears to be cover the same issue, but that is reported as fixed back in v1.0

I have been through the documentation, and there appears to be no way to disable the DJLint formatter (the 'djlint:off' comment appears to only disable linting, not formatting?).

### The issue:

I'm embedding some Jinja2 code inside a script block, and unfortunately the embedded Javascript formatter is butchering the code badly... so I've used the 'beautify preserve:start/end' pragmas to disable the javascript formatting... which works.

However, when I disable the javascript formatting, the djlint formatter starts adding extra indentation on every execution.

Sample code (entire contents of 'formatter_test.html.j2' file):
```
{% macro generate_clear_script(ids) %}
    <script type="text/javascript">
    /* beautify preserve:start */
        function reset_adv_search_form() {
            {%- for id in ids %}
            $('#{{ id }}').val([]);
            $('#{{ id }} option').prop("selected", false);
            {%- endfor %}
            $('#from_date').val('');
        }
    /* beautify preserve:end */
    </script>
{% endmacro %}
```
After the first iteration:
```
{% macro generate_clear_script(ids) %}
    <script type="text/javascript">
        /* beautify preserve:start */
                function reset_adv_search_form() {
                    {%- for id in ids %}
                    $('#{{ id }}').val([]);
                    $('#{{ id }} option').prop("selected", false);
                    {%- endfor %}
                    $('#from_date').val('');
                }
            /* beautify preserve:end */
    </script>
{% endmacro %}
```
After the second iteration:
```
{% macro generate_clear_script(ids) %}
    <script type="text/javascript">
        /* beautify preserve:start */
                        function reset_adv_search_form() {
                            {%- for id in ids %}
                            $('#{{ id }}').val([]);
                            $('#{{ id }} option').prop("selected", false);
                            {%- endfor %}
                            $('#from_date').val('');
                        }
                    /* beautify preserve:end */
    </script>
{% endmacro %}
```
And third iteration:
```
{% macro generate_clear_script(ids) %}
    <script type="text/javascript">
        /* beautify preserve:start */
                                function reset_adv_search_form() {
                                    {%- for id in ids %}
                                    $('#{{ id }}').val([]);
                                    $('#{{ id }} option').prop("selected", false);
                                    {%- endfor %}
                                    $('#from_date').val('');
                                }
                            /* beautify preserve:end */
    </script>
{% endmacro %}
```
I've tested this behaviour up to 15 iterations, at which point I stopped.  Every iteration, the code ends up further and further indented.

Note: Also tested the same code using 'djlint:off/on' comments - had no impact (same indenting behaviour observed).  test code with djlint:off/on comments, for reference:
```
{% macro generate_clear_script(ids) %}
    {# djlint:off #}
    <script type="text/javascript">
    /* beautify preserve:start */
        function reset_adv_search_form() {
            {%- for id in ids %}
            $('#{{ id }}').val([]);
            $('#{{ id }} option').prop("selected", false);
            {%- endfor %}
            $('#from_date').val('');
        }
    /* beautify preserve:end */
    </script>
    {# djlint:on #}
{% endmacro %}
```
Current workaround: I've explicitly added the impacted file to the exclusion list in the .djlintrc file, but this is sub-optimal, as it means that none of the code in the file is validated as part of the CI/CD pipeline.

**Comment by christopherpickering** [comment]
@logicalChimp I've copied your code and cannot reproduce on a mac. Can you provide some details about your OS, and share a complete file where you are able to reproduce this?

**Comment by logicalChimp** [comment]
Hi

Attached is the .djlintrc config file, and the cut-down sample file I was testing with.

In terms of folder structure (in case it makes an impact):

root <folder>
 - .djlintrc
 - app <folder>
   - templates <folder>
     - formatter_test.html.j2

Running from the command line (in the 'root') folder, using the command:
  djlint app/templates --lint --reformat

Using version 1.9.4, managed via pip3 / requirements.txt:
djlint==1.9.4

Hope that's enough for you to replicate the issue.

Cheers

Geoff


________________________________

**Comment by logicalChimp** [comment]
Sigh - many apologies - hit 'send' too quickly.


OS: Ubuntu 20.04.5 LTS
Running in a Python VEnv (so I'm confident that only that specific version of djlint is running) - also verified by running 'djlint --version' on the command line (confirms 1.9.4).

Cheers

Geoff


________________________________

**Comment by christopherpickering** [comment]
Now I can reproduce, thanks :)

**Comment by christopherpickering** [comment]
The problem is coming from the `/* beautify preserve:start/end */` as djlint is assuming that the text inside the script will be formatted with the already existing left indent removed. The preserver tag is keeping it, so when djlint applies the current indent to that section it is doubled.

I can also check for those tags and not add extra indent to text between them. The output will look something like this:

```html
{% macro generate_clear_script(ids) %}
    <script type="text/javascript">
        /* beautify preserve:start */ <<<< note the updated indent here
        function reset_adv_search_form() {
            {%- for id in ids %}
            $('#{{ id }}').val([]);
            $('#{{ id }} option').prop("selected", false);
            {%- endfor %}
            $('#from_date').val('');
        }
        /* beautify preserve:end */ <<<< note the updated indent here
    </script>
{% endmacro %}
```

**Comment by logicalChimp** [comment]
That will be fine.

Thanks

Geoff


________________________________

**Comment by logicalChimp** [comment]
Whilst I remember - the same behaviour was also happening when using the /* beautify ignore:start */ comments too (likely for the same reason)

**Comment by christopherpickering** [comment]
Nice, thanks for pointing that out. It looks like the ignore applies to css and js, while preserve only js.

**Comment by logicalChimp** [comment]
Thanks for being so responsive, and looking into it so quickly :)

When is the next scheduled release?

**Comment by christopherpickering** [comment]
No problem! A few minutes from now hopefully 😀

**Comment by logicalChimp** [comment]
Just retested using 1.12.3 - confirmed fixed.

Many thanks! :)

**Comment by christopherpickering** [comment]
nice, thanks! Sorry about the trouble.
